cmd/signature-v4-utils.go

```diff
@@ -157,3 +157,15 @@ func extractSignedHeaders(signedHeaders []string, reqHeaders http.Header) (http.
 
 	return extractedSignedHeaders, ErrNone
 }
+
+// Sequential spaces regular expression
+var seqSpaces = regexp.MustCompile("  +")
+
+// Trim leading and trailing spaces and replace sequential spaces with one space, following Trimall()
+// in http://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html
+func sigV4TrimAll(input string) string {
+	input = strings.TrimPrefix(input, " ")
+	input = strings.TrimSuffix(input, " ")
```

Comment:
Well, I understand now.. it is aws-java-sdk which is cleaning the header value before sending it to the server.